test/exe/Main.hs

```diff
@@ -3065,6 +3083,8 @@ ifaceErrorTest = testCase "iface-error-test-1" $ withoutStackEnv $ runWithExtraF
     --   - P is being typechecked with the last successful artifacts for A.
     expectDiagnostics [("P.hs", [(DsWarning,(4,0), "Top-level binding")])
                       ,("P.hs", [(DsWarning,(6,0), "Top-level binding")])
+                      ,("P.hs", [(DsInfo,(4,0), "Defined but not used")])
```

Comment:
The really important thing is that these diagnostics have the right tag associated with them. While they don't all need to be `expectDiagnostics_` (or whatever it ends up being called), a few more being that with the tag assertion would be good.

Reply:
That makes sense. I've extended the checks in the latest commit.